brain-grammar/src/grammar/expressions/map.rs

```diff
@@ -47,13 +47,21 @@ impl Parse for Map {
         while !stream.check(BrainToken::RightBrace) {
             let mut key = Expression::parse(stream)?;
 
-            if let Expression::Identifier(identifier) = key {
-                key = Expression::new_literal(Value::String(identifier.name));
-            } else {
-                return Err(Error::new(
-                    ErrorKind::UnexpectedExpression,
-                    format!("Only string values may be keys."),
-                ));
+            match key {
```

Comment:
you shouldn't do any evaluating in the parse section. you can leave evaluation to the evaluate section and it should naturally resolve to a value. we just need to make sure that parse can accept a literal which does resolve to a value here too.